crates/ruff_linter/resources/test/fixtures/pydoclint/DOC101_google.py

```diff
@@ -0,0 +1,372 @@
+        'location': location
+    }
+
+
+# DOC101
+def calculate_total_price(item_prices, tax_rate, discount):
+    """
+    Calculates the total price after applying tax and a discount.
+
+    Args:
+        item_prices (list of float): A list of prices for each item.
+
+    Returns:
+        float: The final total price after tax and discount.
+    """
+    total = sum(item_prices)
+    total_with_tax = total + (total * tax_rate)
+    final_total = total_with_tax - discount
+    return final_total
+
+
+# DOC101
+def send_email(subject, body, to_address, cc_address=None, bcc_address=None):
+    """
+    Sends an email to the specified recipients.
+
+    Args:
+        subject (str): The subject of the email.
+        body (str): The content of the email.
+        to_address (str): The recipient's email address.
+
+    Returns:
+        bool: True if the email was sent successfully, False otherwise.
+    """
+    return True
+
+
+# DOC101
+def concatenate_strings(separator, *args):
+    """
+    Concatenates multiple strings with a specified separator.
+
+    Args:
+        separator (str): The separator to use between strings.
+
+    Returns:
+        str: A single concatenated string.
+    """
+    return separator.join(args)
+
+
+# DOC101
+def process_order(order_id, *items, **details):
+    """
+    Processes an order with a list of items and optional order details.
+
+    Args:
+        order_id (int): The unique identifier for the order.
+
+    Returns:
+        dict: A dictionary containing the order summary.
+    """
+    return {
+        'order_id': order_id,
+        'items': items,
+        'details': details
+    }
+
+
+class Calculator:
+    """
+    A simple calculator class that can perform basic arithmetic operations.
+    """
+
+    # DOC101
+    def __init__(self, value=0):
+        """
+        Initializes the calculator with an initial value.
+
+        Returns:
+            None
+        """
+        self.value = value
+
+    # DOC101
+    def add(self, number, number2):
+        """
+        Adds a number to the current value.
+
+        Args:
+            number (int or float): The number to add to the current value.
+
+        Returns:
+            int or float: The updated value after addition.
+        """
+        self.value += number + number2
+        return self.value
+
+    # DOC101
+    @classmethod
+    def from_string(cls, value_str):
+        """
+        Creates a Calculator instance from a string representation of a number.
+
+        Returns:
+            Calculator: A new instance of Calculator initialized with the value from the string.
+        """
+        value = float(value_str)
+        return cls(value)
+
+    # DOC101
+    @staticmethod
+    def is_valid_number(number):
+        """
+        Checks if a given number is valid (int or float).
+
+        Returns:
+            bool: True if the number is valid, False otherwise.
+        """
+        return isinstance(number, (int, float))
```

Comment:
What happens if the method has overloads? Is it required to document all parameters for each overload? What about other abstract methods?

Reply:
There is already a discussion to optionally ignore those type of functions (https://github.com/astral-sh/ruff/issues/12434#issuecomment-2305425941)

But I think it should be a seperate PR.